dome9CloudBots/bots/delete_network_security_group_single_rule.py

```diff
@@ -0,0 +1,119 @@
+# What it does: Delete network security group rule
+# Deletion will be preformed by given destination port, destination scope, source port, source scope and access
+# Access can be : Allow or Deny
+# Usage: AUTO: delete_network_security_group_single_rule <destination port> <destination scope> <source port> <source scope> <access>
+# Example: AUTO: delete_network_security_group_single_rule 556 10.0.0.0/24 22 0.0.0.0/2 Allow
+# In case you don't want to use specific destination you need to fill '-' in the specific field.
+# Example 1: AUTO: delete_network_security_group_single_rule 556 10.0.0.0/24 - - Allow
+# Example 2: AUTO: delete_network_security_group_single_rule - - 22 0.0.0.0/2 Deny
+# Limitations: None
+
+from msrestazure.azure_exceptions import CloudError
+from azure.mgmt.network import NetworkManagementClient
+import logging
+
+ONE_PORT = 1
+
+
+def is_port_range(port):
+    return len(port) > ONE_PORT
+
+
+def is_port_in_range(port_to_find, ports_list):
+    for port in ports_list:
+        if port == port_to_find:
+            return True
+
+        else:
+            # Due to Azure SDK we split cases as port can be : '360' or '360-366'
+            ports = port.split('-')
+            starting_port, ending_port = ports
+            if is_port_range(ports) and port_to_find > starting_port and port_to_find < ending_port:
+                return True
+
+    return False
+
+
+def is_port_match(rule, direction, port):
+    port_to_find = rule.get(f'{direction}_port_range')
+    ports_to_find = rule.get(f'{direction}_port_ranges')
+    if port_to_find:
+        if port_to_find == port:
+            return True
+        else:
+            ports = port_to_find.split('-')
+            if (len(ports) > ONE_PORT):
+                starting_port, ending_port = ports
+                if is_port_range(ports) and port > starting_port and port < ending_port:
+                    return True
+    elif ports_to_find:
+        if is_port_in_range(port, ports_to_find):
+            return True
+    return False
+
+
+def is_scope_match(rule, direction, scope):
+    return rule[f'{direction}_address_prefix'] == scope
+
+
+def is_access_match(rule, access):
+    return rule['access'] == access
+
+
+def is_rule_should_be_deleted_by_direction(rule, direction, port, scope, access):
+    return is_port_match(rule, direction, port) and is_scope_match(rule, direction, scope) and is_access_match(rule,
+                                                                                                               access)
+
+
+def is_rule_should_be_deleted(rule, destination_port, destination_scope, source_port, source_scope, access):
+    ## Case need to check both destination and source
+    if (destination_port != '-' and destination_scope != '-' and source_port != '-' and source_scope != '-'):
```

Comment:
if all 
https://stackoverflow.com/questions/3844801/check-if-all-elements-in-a-list-are-identical